### PR TITLE
Implemented K'rrik, Son of Yawgmoth

### PR DESCRIPTION
**Pull request by ssouders412**
[FIXED in 84701cf] KNOWN ISSUE: K'rrik's ability allowing the player to pay Phyrexian for black mana remains active after K'rrik leaves the battlefield.

### CODE REVIEW

File: Mage/src/main/java/mage/players/PlayerImpl.java [hunk 1]
```diff
@@ -4160,4 +4165,48 @@ public int hashCode() {
...
+            this.phyrexianColors.setRed(false);
+        }
+        if (colors.isGreen()) {
+            this.phyrexianColors.setGreen(false);
+        }
+    }
+    
+    @Override
+    public FilterMana getPhyrexianColors() {
+        return this.phyrexianColors.copy();
```
Comment (JayDi85) [inline Mage/src/main/java/mage/players/PlayerImpl.java]: Why it's makes copies in every getter call instead simple ref? Normal code: return ref objects and create copy on Copy/Create method (but you are using ref in copy -- see above).

File: Mage/src/main/java/mage/players/PlayerImpl.java [hunk 2]
```diff
@@ -276,6 +279,8 @@ public PlayerImpl(final PlayerImpl player) {
         this.payManaMode = player.payManaMode;
 
         this.designations.addAll(player.designations);
+        
+        this.phyrexianColors = player.phyrexianColors;
```
Comment (JayDi85) [inline Mage/src/main/java/mage/players/PlayerImpl.java]: You must copy all changeable objects (do not use refs).

**Comment by JayDi85** [comment]
Thanks for PR.

**Comment by ssouders412** [comment]
#5923

**Pull request merged**